openshift/catalog-sync/pkg/check/image_check.go

```diff
@@ -0,0 +1,79 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"oras.land/oras-go/v2"
+	"os"
+	"path/filepath"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type Checks struct {
+	ImageChecks      []ImageCheck
+	FilesystemChecks []FilesystemCheck
+	CatalogChecks    []CatalogCheck
+}
+
+type ImageCheckFunc func(ctx context.Context, root ocispec.Descriptor, target oras.ReadOnlyTarget) error
+type FilesystemCheckFunc func(ctx context.Context, imageFS fs.FS) error
+type CatalogCheckFunc func(ctx context.Context, fbcFS fs.FS) error
```

Comment:
This should probably change to:
```suggestion
type CatalogCheckFunc func(ctx context.Context, declcfg.DeclarativeConfig) error
```

That way, we can have our framework convert to that type once, and then all catalog-based checks don't have to repeat this (costly) step.